src/Http/Driver/CurlDriver.php

```diff
@@ -77,7 +76,9 @@ public function execute(RequestInterface $request)
         }
 
         $response = new Response();
-        $response->withStatus($info['http_code']);
+
+        $response = $response->withStatus($info['http_code']);
```

Comment:
Would `http_code` have  ever be set before? `->with*` methods are not touching the immutable instance, right?